### PR TITLE
[backend] Customize time range of relationship de-deduplication (#3387)

### LINKED ISSUE #3387: Be able to customize the time range of relationship de-deduplication mechanism
## Use case

Be able to customize the time range of relationship de-deduplication mechanism.
Be able to add (as an option) the author as a deduplication key.

Currently, we deduplicate relationships on:

- source ID
- target ID
-  -1 month < start_time  < +1 month
- -1 month < stop_time  < +1 month

We need to propose the ability to change the resolved range for de-deplucation.

- [ ] Add static configuration in platform config
- [ ] Adapt engine algorithm to take into account the configuration

This will enable a powerful capability for the platform to be more "precise" as a threat intelligence team but also to de-duplicate everything based on 12 months range for instance for less mature / less resourced team that just need the "higher level" insights.

### PR DESCRIPTION
**Pull request by richard-julien**
See #3387

### CODE REVIEW

File: opencti-platform/opencti-graphql/config/default.json [hunk 1]
```diff
@@ -91,6 +91,11 @@
   "no_proxy": "",
   "https_proxy_ca": [],
   "https_proxy_reject_unauthorized": false,
+  "relations_deduplication": {
+    "past_days": 30,
+    "next_days": 30,
+    "created_by_based": false
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/config/default.json:97]: question: what does `created_by_based` mean ?

**Review by aHenryJard — COMMENTED** [review]
Looks good to my newbie eyes, but CI is failing, and I wonder if it's possible to have a test coverage for this feature or if the test already exists.

File: opencti-platform/opencti-graphql/config/default.json (hunk 1 above)
Comment (richard-julien) [reply opencti-platform/opencti-graphql/config/default.json:97]: created_by_based is used to add the createdBy information as part of the deduplication query.

JRI creates a relationship TARGETS, with FROM_DATE1 and FROM_DATE2 will not be dedupe if SAM creates relationship TARGETS, with FROM_DATE1 and FROM_DATE2, just creating a new one

Will take a look to the tests

Comment (richard-julien) [reply opencti-platform/opencti-graphql/config/default.json:97]: Tests are now running fine

**Review by aHenryJard — APPROVED** [review]

**Pull request merged**